tests/Backend/CommonPart1/CreateTableTest.php

```diff
@@ -24,6 +24,89 @@ public function setUp()
         $this->_initEmptyTestBuckets();
     }
 
+    public function testSyntheticPrimaryKey()
+    {
+        $testBucketName = $this->getTestBucketName(__METHOD__);
+        $testBucketStage = self::STAGE_IN;
+        $testBucketId = $testBucketStage . '.c-' . $testBucketName;
+        $tableName = 'testSynthPk';
+        $testTableId = $testBucketId . '.' . $tableName;
+
+        $this->dropBucketIfExists($this->_client, $testBucketId);
+        $testBucketId = $this->_client->createBucket($testBucketName, self::STAGE_IN);
+        $this->_client->createTable(
+            $testBucketId,
+            $tableName,
+            new CsvFile(__DIR__ . '/../../_data/languages.csv')
+        );
+
+        // test spk part of detail
+        $table = $this->_client->getTable($testBucketId . '.' . $tableName);
+        $this->assertSame(false, $table['syntheticPrimaryKeyEnabled']);
+
+        try {
+            $tableId = $this->_client->createTable(
+                $testBucketId,
+                'myTable',
+                new CsvFile(__DIR__ . '/../../_data/languages.csv'),
+                ['syntheticPrimaryKeyEnabled' => 1, 'primaryKey' => 'id,name']
+            );
+            $this->fail('Should have thrown');
+        } catch (ClientException $e) {
+            $this->assertSame(
+                'Synthetic primary key cannot be enabled in sync call, use async call to create the table',
+                $e->getMessage()
+            );
+        }
+
+        try {
+            $tableId = $this->_client->createTableAsync(
+                $testBucketId,
+                'myTable',
+                new CsvFile(__DIR__ . '/../../_data/languages.csv'),
+                ['syntheticPrimaryKeyEnabled' => 1]
+            );
+            $this->fail('Should have thrown');
+        } catch (ClientException $e) {
+            $this->assertSame('Cannot set synthetic primary key if primary key is empty', $e->getMessage());
```

Comment:
Teď je to asi OK, když nemáme PK add remove, ale každopádně by to pak mělo jít zapnout pro tabulky bez PK.

Reply:
A z čeho to udělá ten hash? Ze všeho?

Reply:
No neudělá, ale je to v metadatech, takže když pak uděláš add PK tak ti zrovna vytvoří SPK

Reply:
Ale nemá to význam dokud neuděláme přidání odebrání PK, imho

Reply:
To mi přijde divné chování. Z metadat vykoukáš, že máš na té tabulce spk, ale buď chybí ten sloupec nebo v něm nejsou data. To mi přijde jako fail.

Reply:
Je to tady hned první věta: https://keboola.atlassian.net/browse/KBC-493
```
U tabulky půjde zapnout, že se generuje syntetický PKEY, takže jakmile bude mít sloupeček s víc jak 1 PKEY, tak KBC dodá sloupec (třeba) _pkey, který vznikne jako
```
jen sa změnilo to, že to bude generovat SPK už při jedno sloupci.

Reply:
Nerozporuju, že je to v zadání, ale že mi to přijde blbě. Když by to šlo zapnout na tabulce bez definice klíčů, tak by to znamenalo, že abys zjistil jestli tam ten SPK sloupec je nebo ne, tak bys musel udělat describe té tabulky na backendu. Ani bys nemohl použít metadata ke sloupcům tabulky - protože tam to být nemůže - jako tam není timestamp.

Reply:
No možem to probrat na ofsajtu třeba. Každopádně to teď není blocker a zatím to tak može byt. Jinak v tom nevidím problém, to jestli je tam SPK víš z metadat protože když je SPK zapnuté a nemáš žádný PK, tak tam ten SPK sloupec není, pokud máš SPK zapnuté a je tak 1+PK tak tam SPK musí být.

Reply:
OK